Source/WebCore/platform/graphics/PlatformDisplay.cpp

```diff
@@ -331,6 +347,64 @@ void PlatformDisplay::terminateEGLDisplay()
     m_eglDisplay = EGL_NO_DISPLAY;
 }
 
+EGLImage PlatformDisplay::createEGLImage(EGLContext context, EGLenum target, EGLClientBuffer clientBuffer, const Vector<EGLAttrib>& attributes) const
+{
+    if (eglCheckVersion(1, 5)) {
+#if USE(LIBEPOXY)
+        return eglCreateImage(m_eglDisplay, context, target, clientBuffer, attributes.data());
+#else
+        static PFNEGLCREATEIMAGEPROC s_eglCreateImage = reinterpret_cast<PFNEGLCREATEIMAGEPROC>(eglGetProcAddress("eglCreateImage"));
+        if (s_eglCreateImage)
+            return s_eglCreateImage(m_eglDisplay, context, target, clientBuffer, attributes.data());
+        return EGL_NO_IMAGE;
+#endif
+    }
+
+    if (!m_eglExtensions.KHR_image_base)
+        return EGL_NO_IMAGE;
+
+    Vector<EGLint> intAttributes;
+    if (!attributes.isEmpty()) {
+        intAttributes = attributes.map<Vector<EGLint>>([] (EGLAttrib value) {
+            return value;
+        });
+    }
+#if USE(LIBEPOXY)
+    return eglCreateImageKHR(m_eglDisplay, context, target, clientBuffer, intAttributes.data());
+#else
+    static PFNEGLCREATEIMAGEKHRPROC s_eglCreateImageKHR = reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(eglGetProcAddress("eglCreateImageKHR"));
+    if (s_eglCreateImageKHR)
+        return s_eglCreateImageKHR(m_eglDisplay, context, target, clientBuffer, intAttributes.data());
+    return EGL_NO_IMAGE_KHR;
+#endif
+}
```

Comment:
When passing the attributes Vector data to the create-image call, the Vector's data pointer should be passed if the Vector is not empty, and nullptr otherwise. This avoids breaking things through zero-sized Vector with an allocated buffer.

The mapping into EGLint can then also be done without an is-empty check.

Reply:
Ok.